cassdegrees/ui/views.py

```diff
@@ -31,11 +32,71 @@ def planList(request):
     :param request:
     :return <class django.http.response.HttpResponse>:
     """
-    degree = requests.get(request.build_absolute_uri('/api/model/degree/?format=json')).json()
-    subplan = requests.get(request.build_absolute_uri('/api/model/subplan/?format=json')).json()
-    course = requests.get(request.build_absolute_uri('/api/model/course/?format=json')).json()
+    query = request.GET.get('q', '')
 
-    return render(request, 'list.html', context={'data': {'Degree': degree, 'Subplan': subplan, 'Course': course}})
+    # No search, render default page
+    if not query:
+        degree = requests.get(request.build_absolute_uri('/api/model/degree/?format=json')).json()
```

Comment:
Is there a reason you have gone down the path of implementing search manually, rather than using a pre-built solution like https://fusejs.io/, which would likely do a few more things here (e.g. fuzzy search)?

Not that this is bad in itself though.

Reply:
Admittedly I could have done a bit more research in to existing tools. Initially it was performing a much more basic search which wasn't too bad on the computation/technical front but the complexity grew once I decided to prioritise search results.

Reply:
It doesn't particularly matter. if you do get time, fuzzy search would be nice here (i.e for typos), but other than that, LGTM!